File: JAVA/java/src/IPO.java
import java.util.PriorityQueue;

public class IPO {
    public static int findMaximizedCapital(int k, int W, int[] Profits, int[] Capital) {
        PriorityQueue<int[]> pqCap = new PriorityQueue<>((a, b) -> (a[0] - b[0]));
        PriorityQueue<int[]> pqPro = new PriorityQueue<>((a, b) -> (b[1] - a[1]));

        for (int i = 0; i < Profits.length; i++) {
            pqCap.add(new int[] {Capital[i], Profits[i]});
        }

        for (int i = 0; i < k; i++) {
            while (!pqCap.isEmpty() && pqCap.peek()[0] <= W) {
                pqPro.add(pqCap.poll());
            }

            if (pqPro.isEmpty()) break;

            W += pqPro.poll()[1];
        }

        return W;
    }

    public static void main(String[] args) {
        System.out.println(findMaximizedCapital(2, 0,new int[]{1,2,3},  new int[]{0,1,1}));

    }
}

File: python/IPO.py
import heapq
def findMaximizedCapital(
         k: int, w: int, profits, capital
) -> int:
    n = len(profits)
    projects = [(capital[i], profits[i]) for i in range(n)]
    projects.sort()
    maxHeap = []
    i = 0
    for _ in range(k):
        while i < n and projects[i][0] <= w:
            heapq.heappush(maxHeap, -projects[i][1])
            i += 1
        if not maxHeap:
            break
        w -= heapq.heappop(maxHeap)

    return w
print(findMaximizedCapital(2,  0,[1,2,3],[0,1,1]))